test/integration/volume/persistent_volumes_test.go

```diff
@@ -945,16 +953,17 @@ func TestPersistentVolumeProvisionMultiPVCs(t *testing.T) {
 // TestPersistentVolumeMultiPVsDiffAccessModes tests binding of one PVC to two
 // PVs with different access modes.
 func TestPersistentVolumeMultiPVsDiffAccessModes(t *testing.T) {
-	_, s, closeFn := framework.RunAnAPIServer(nil)
-	defer closeFn()
+	s := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{"--disable-admission-plugins=ServiceAccount,StorageObjectInUseProtection"}, framework.SharedEtcd())
+	defer s.TearDownFn()
+	namespaceName := "multi-pvs-diff-access"
 
-	ns := framework.CreateTestingNamespace("multi-pvs-diff-access", t)
-	defer framework.DeleteTestingNamespace(ns, t)
-
-	testClient, controller, informers, watchPV, watchPVC := createClients(ns, t, s, defaultSyncPeriod)
+	testClient, controller, informers, watchPV, watchPVC := createClients(namespaceName, t, s, defaultSyncPeriod)
 	defer watchPV.Stop()
 	defer watchPVC.Stop()
 
+	ns := framework.CreateNamespaceOrDie(testClient, namespaceName, t)
+	defer framework.DeleteNamespaceOrDie(testClient, ns, t)
```

Comment:
you always create the namespace after create the clients in this commit

> 	watchPVC, err := testClient.CoreV1().PersistentVolumeClaims(namespaceName).Watch(context.TODO(), metav1.ListOptions{})

I think that doesn't matter, but is there any special reason?

Reply:
Before this PR we effectively weren't creating the namespace (CreateTestingNamespace isn't effectively creating a namespace:
https://github.com/kubernetes/kubernetes/blob/master/test/integration/framework/util.go#L54

So we didn't need the client for that.

After switching to more realistic apiserver in this PR (that e.g. has namespace admission plugins enabled) we have to actually create a namespace (or alternatively disable namespace admission plugins, but I think it's better to those test be as realistic as possible), which requires having a client.

So I need to do that after creating a client.

Now the options are:
- refactor the code by splitting client creation and setting up those watches (and create namespaces in the middle)
- leave as is and create namespace after calling `createClients`

Since this doesn't matter really, I chose a cheaper to execute option :)